### PR TITLE
APP-28934: feat(Select): add openOnHover prop

### PR DESCRIPTION
**Pull request by habxtech**
See [APP-28934](https://habxfr.atlassian.net/browse/APP-28934): Add openOnHover prop on Select.

### CODE REVIEW

File: src/Menu/Menu.style.ts [hunk 1]
```diff
@@ -9,15 +9,17 @@ const menuCssVariables = css`
 
 export const FloatingMenuContainer = styled.div`
   height: 100%;
-  box-shadow: ${theme.shadow()};
-  border-radius: 4px;
+  padding: 8px 0;
```
Comment (jean9696) [inline src/Menu/Menu.style.ts:12]: Why did you change the style ?

File: src/Menu/Menu.tsx [hunk 2]
```diff
@@ -63,6 +64,8 @@ export const Menu = React.forwardRef<HTMLDivElement, MenuProps>(
         fullScreenOnMobile={fullScreenOnMobile}
         ref={ref}
         setStyle={setStyle}
+        withOverlay={withOverlay}
+        simulated
```
Comment (jean9696) [inline src/Menu/Menu.tsx:68]: Why do you need `simulated` ?

File: src/Menu/Menu.tsx [hunk 3]
```diff
@@ -22,6 +22,7 @@ export const Menu = React.forwardRef<HTMLDivElement, MenuProps>(
       position = 'vertical-stretched',
       scrollable,
       setPosition = menuDefaultPositionSetter,
+      withOverlay,
```
Comment (jean9696) [inline src/Menu/Menu.tsx:25]:
```suggestion
```

File: src/Menu/Menu.tsx [hunk 4]
```diff
@@ -63,6 +64,8 @@ export const Menu = React.forwardRef<HTMLDivElement, MenuProps>(
         fullScreenOnMobile={fullScreenOnMobile}
         ref={ref}
         setStyle={setStyle}
+        withOverlay={withOverlay}
```
Comment (jean9696) [inline src/Menu/Menu.tsx:67]:
```suggestion
```

File: src/Menu/Menu.utils.ts [hunk 5]
```diff
@@ -73,11 +73,11 @@ export const menuDefaultPositionSetter: MenuPositionSetter = ({
     case 'vertical-stretched': {
       let minWidth: number | undefined
 
-      const bottomOfTrigger = triggerDimensions.bottom + MARGIN
+      const bottomOfTrigger = triggerDimensions.bottom
```
Comment (jean9696) [inline src/Menu/Menu.utils.ts:76]: again this PR should not about styling 🤔

File: src/Menu/Menu.style.ts (hunk 1 above)
Comment (lweinhard) [reply src/Menu/Menu.style.ts:12]: `box-shadow` and `border-radius` didn't belong here.
As for `padding` goes, there was a slight space between `Options` and `Select` components that was preventing `Options` to stay opened when hovering.
Thus explaining why I needed to add a little extra space.

File: src/Select/Select.hooks.ts [hunk 6]
```diff
@@ -147,12 +152,21 @@ export const useSelect = ({
     []
   )
 
-  const handleOpen = React.useCallback(() => {
-    dispatch({ type: ActionType.Open })
+  const handleClick = React.useCallback(() => {
+      dispatch({ type: ActionType.Focus })
```
Comment (jean9696) [inline src/Select/Select.hooks.ts]: Why Focus doesn't open in reducer ?

File: src/Select/Select.interface.ts [hunk 7]
```diff
@@ -59,6 +60,7 @@ export interface SelectProps extends WithLabel<SelectInnerProps> {}
 
 export interface SelectState {
   isOpened: boolean
+  isFocused: boolean
```
Comment (jean9696) [inline src/Select/Select.interface.ts:63]: `shouldn't it be `isHovered` ?

File: src/Select/Select.tsx [hunk 8]
```diff
@@ -53,13 +54,17 @@ const InnerSelect = React.forwardRef<HTMLDivElement, SelectInnerProps>(
...
       value,
+      openOnHover,
     })
 
     const context = React.useMemo<SelectContextValue>(
       () => ({ multi, canSelectAll }),
       [multi, canSelectAll]
     )
 
+    const onMouseOver = actions.onHover
```
Comment (jean9696) [inline src/Select/Select.tsx]: Why don't you check the `openOnHover` here ?

File: src/Select/Select.tsx [hunk 9]
```diff
@@ -109,25 +119,31 @@ const InnerSelect = React.forwardRef<HTMLDivElement, SelectInnerProps>(
                 <Line />
               </ResetIconContainer>
             )}
-            <ElementRightContainer>
+            <ElementRightContainer onMouseOver={(e) => {
+              e.stopPropagation()
```
Comment (jean9696) [inline src/Select/Select.tsx:122]: Why stopping propagation ?

**Review by jean9696 — CHANGES_REQUESTED** [review]

File: src/Select/Select.tsx (hunk 8 above)
Comment (lweinhard) [reply src/Select/Select.tsx]: Because it is already checked inside the reducer.

File: src/Select/Select.interface.ts (hunk 7 above)
Comment (lweinhard) [reply src/Select/Select.interface.ts:63]: I first started to implement it along with `isFocused` but I ended up noticing that I didn't need it.

File: src/Select/Select.hooks.ts (hunk 6 above)
Comment (lweinhard) [reply src/Select/Select.hooks.ts]: It does, the line right below is another dispatch of `Open` action type

File: src/Menu/Menu.utils.ts (hunk 5 above)
Comment (lweinhard) [reply src/Menu/Menu.utils.ts:76]: This is related to the changes I needed to operate in order for options to stay opened while hovering.

File: src/Menu/Menu.tsx (hunk 3 above)
Comment (lweinhard) [reply src/Menu/Menu.tsx:25]: I need to remove that overlay if `openOnHover` is true.

File: src/Menu/Menu.tsx (hunk 4 above)
Comment (lweinhard) [reply src/Menu/Menu.tsx:67]: I need to remove that overlay if openOnHover is true.

File: src/Select/Select.tsx (hunk 9 above)
Comment (lweinhard) [reply src/Select/Select.tsx:122]: Because I didn't want it to open `Select` using `onMouseOver` from parent.

File: src/Menu/Menu.tsx (hunk 2 above)
Comment (lweinhard) [reply src/Menu/Menu.tsx:68]: I need to pass parent background to child.

File: src/Select/Select.hooks.ts (hunk 6 above)
Comment (jean9696) [reply src/Select/Select.hooks.ts]: Yes but why doing in with 2 dispatch instead of one ?

File: src/Select/Select.interface.ts (hunk 7 above)
Comment (jean9696) [reply src/Select/Select.interface.ts:63]: But it's not really focused 🤔

File: src/TogglePanel/TogglePanel.interface.ts [hunk 10]
```diff
@@ -21,6 +22,8 @@ export interface InnerTogglePanelProps
   setStyle?: TogglePanelStyleSetter
   triggerRef?: React.RefObject<HTMLElement | null>
   withOverlay?: boolean
+  backgroundColor?: ColorGetter
```
Comment (jean9696) [inline src/TogglePanel/TogglePanel.interface.ts]: You could maybe just extend `BackgroundProps` ?

File: src/Select/Select.tsx (hunk 8 above)
Comment (jean9696) [reply src/Select/Select.tsx]: That's weird to check this behaviour in 2 places don't you think ?

File: src/Select/Select.interface.ts (hunk 7 above)
Comment (lweinhard) [reply src/Select/Select.interface.ts:63]: `isFocused` allows to track whether or not you clicked to open the component.
If you have a better name, I'll take it.

Comment (jean9696) [reply src/Select/Select.interface.ts:63]: Oh I see, I misunderstood what you meant. `isFocused` is alright

**Review by jean9696 — APPROVED** [review]
